cmd/kube-rbac-proxy/app/options/proxyoptions.go

```diff
@@ -116,29 +120,30 @@ func (o *ProxyOptions) Validate() []error {
 	return errs
 }
 
-func (o *ProxyOptions) ApplyTo(c *server.KubeRBACProxyInfo, a *serverconfig.AuthenticationInfo) error {
+func (o *ProxyOptions) ApplyTo(config *server.KubeRBACProxyConfig) error {
```

Comment:
The previous was better in terms of isolation.

Reply:
Yes, I know, but I need it to set up HTTP/2 disablement. We can revert it in k8s 1.31.

Reply:
No, we set more than the secure serving, we also have the krp info and delegated authn that contains authn / authz and routing information..

Reply:
I meant on top of all the other things that were being passed in previously

Reply:
Yea, but like 3 arguments? Why not handing in the whole object then?

Reply:
to isolate the changes, and also to stress that the function probably does more things that it should and we may want to refactor it in the near future

Reply:
Yes yes, lets keep it ugly, such that feel the itch to make it better soon.

Reply:
Sorry, I must have dropped it with git.